### PR TITLE
[release-4.8]Bug 1971188: Creating a VM from the UI on OKD 4.7 fails

### PR DESCRIPTION
**Pull request by glekner**
manual cherry pick of 1967885 fix. #9258

### CODE REVIEW

**Comment by yaacov** [comment]
/retest

@vojtechszocs @rawagner hi, this PR should fix https://bugzilla.redhat.com/show_bug.cgi?id=1971188 

Without this PR if you have CNV 2.6.5 (VM apiVersion v1alpha3) you will not see the VMs, so it's very important to get this into 4.8.

File: frontend/packages/console-shared/src/utils/owner-references.ts
Comment (spadgett) [inline frontend/packages/console-shared/src/utils/owner-references.ts:19]: I'm confused why this change was needed. `obj.kind` should always be the k8s kind and not the internal console reference using `~`.

Comment (glekner) [reply frontend/packages/console-shared/src/utils/owner-references.ts:19]: from what I tested there are usages where the kind does contain `~`
it is a needed safety net that doesn't hurt

**Comment by yaacov** [comment]
@vojtechszocs @rawagner PTAL

File: frontend/packages/console-shared/src/utils/owner-references.ts
Comment (spadgett) [reply frontend/packages/console-shared/src/utils/owner-references.ts:19]: What were the usages? That sounds like a bug.

File: frontend/packages/kubevirt-plugin/src/components/vms/vm-create-yaml.tsx
Comment (vojtechszocs) [inline frontend/packages/kubevirt-plugin/src/components/vms/vm-create-yaml.tsx:95]: Compared to #9258 this is missing prop

```
kindObj={kubevirtReferenceForModel(VirtualMachineModel)}
```

so please double-check, otherwise looks good.

**Comment by spadgett** [comment]
/hold
I'd like to better understand the change for the owner references check before merging.

**Comment by yaacov** [comment]
@spadgett hi, I see that @glekner removed the changes in `owner-references.ts` can you take a look ?

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**
